### PR TITLE
PLAT-126343:  Add UI tests for agate/Dropdown

### PR DESCRIPTION
**Pull request by stanca-pop-lgp**
Enact-DCO-1.0-Signed-off-by: Stanca Pop stanca.pop@lgepartner.com

### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
- added UI tests for agate/Dropdown

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
PLAT-126343

### Comments

### CODE REVIEW

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM

**Review by adrian-cocoara-lgp — APPROVED** [review]

File: tests/ui/apps/Dropdown/Dropdown-View.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		<Dropdown id="dropdownDefault" title="Default" style={{display: 'inline-block', width: '50%'}}>
+			{['one', 'two', 'three', 'four', 'five']}
+		</Dropdown>
+		<Dropdown direction="right" id="dropdownDirectionRight" title="DirectionRight" style={{display: 'inline-block', width: '50%'}}>
+			{['one', 'two', 'three', 'four', 'five']}
+		</Dropdown>
+		<Dropdown disabled id="dropdownDisabled" title="Disabled">
+			{['one', 'two', 'three', 'four', 'five']}
+		</Dropdown>
+		<Dropdown id="dropdownSelected" defaultSelected={1} title="Selected">
```
Comment (MikyungKim) [inline tests/ui/apps/Dropdown/Dropdown-View.js:21]: I think this missed from the spec. Are you going to test this component too?

File: tests/ui/specs/Dropdown/Dropdown-specs.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+	describe('focus management', function () {
+
+		it('should focus the `#dropdownDefault` when page loads', function () {
+			expect(Page.components.dropdownDefault.childItem.isFocused()).to.be.true();
+		});
+	});
+
+	describe('5-way', function () {
+		beforeEach(function () {
```
Comment (MikyungKim) [inline tests/ui/specs/Dropdown/Dropdown-specs.js:18]: I think we don't need this..

File: tests/ui/specs/Dropdown/Dropdown-specs.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+			Page.spotlightSelect();
+			browser.pause(1000);
+			Page.spotlightDown();
+
+			expect(Page.components.dropdownDefault.item(0).isFocused()).to.be.true();
+		});
+	});
+
+	describe('pointer', function () {
+		beforeEach(function () {
```
Comment (MikyungKim) [inline tests/ui/specs/Dropdown/Dropdown-specs.js:48]: This one also.

File: tests/ui/specs/Dropdown/Dropdown-specs.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+			wrapper.click({x: 0, y: 0});
+
+			browser.pause(1000);
+
+			// Verify that the floating list no longer exists (Dropdown is closed)
+			expect(dropdown.isOpen).to.not.be.true();
+		});
+	});
+
+	describe('in scroller', function () {
```
Comment (MikyungKim) [inline tests/ui/specs/Dropdown/Dropdown-specs.js:71]: How about we separate the spec for `in scroller`?
I think you could refer to FixedPopupPanels from sandstone.

File: tests/ui/specs/Dropdown/Dropdown-specs.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+			Page.spotlightDown();
+			expect(Page.components.dropdownDisabled.childItem.isFocused()).to.be.true();
+		});
+
+		it('should focus the first item in `#dropdownDefault` option list when 5-way enter and down', function () {
+			expect(Page.components.dropdownDefault.childItem.isFocused()).to.be.true();
+
+			Page.spotlightSelect();
+			browser.pause(1000);
```
Comment (MikyungKim) [inline tests/ui/specs/Dropdown/Dropdown-specs.js:40]: I'm not sure this is good. 🤔

File: tests/ui/specs/Dropdown/Dropdown-specs.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+			Page.open('InScroller');
+		});
+
+		function getDropdownOffset (dropdown, scroller) {
+			return browser.execute((a, b) => {
+				return a.getBoundingClientRect().top - b.getBoundingClientRect().top;
+			}, dropdown, scroller);
+		}
+
+		it('should have title visible when focusing button via 5-way', function () {
```
Comment (MikyungKim) [inline tests/ui/specs/Dropdown/Dropdown-specs.js:82]: Is this case related to the dropdown itself? I think it's for scroller...

```suggestion
		it('should scroll into view when navigating dropdown via 5-way', function () {
```

**Review by MikyungKim — CHANGES_REQUESTED** [review]
Good! Please find my comments.
And I think we need more test cases for selecting values.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Comment by jeonghee27** [comment]
I feel that the verification script is somewhat lacking.
If there is a disabled element, it would be better if there is script to verify disabled function.
Hopefully the test script will be added after Agate TC is added more.

**Pull request merged**